### PR TITLE
tests: remove dependency on snapcraft for integration tests

### PR DESCRIPTION
**Pull request by sergiusens**
The integration tests import snapcraft into the suite, this causes all
sorts of problems, the most concerning one is test independence.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [ ] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: tools/travis/run_integration_test.sh [hunk 1]
```diff
@@ -76,6 +76,7 @@ $lxc file push --recursive "$project_path" test-runner/root/
 $lxc exec test-runner -- sh -c "apt install -y bzr git libnacl-dev libssl-dev libsodium-dev libffi-dev libapt-pkg-dev mercurial python3-pip subversion sudo snapd"
 $lxc exec test-runner -- sh -c "python3 -m pip install -I pip==9.0.3"
 $lxc exec test-runner -- sh -c "python3 -m pip install -r snapcraft/requirements-devel.txt -r snapcraft/requirements.txt"
+$lxc exec test-runner -- sh -c "rm -rf snapcraft/snapcraft"
```
Comment (kyrofa) [inline tools/travis/run_integration_test.sh]: Might be worth adding a comment here as to why we're doing this.

File: tests/integration/plugins_nodejs/test_nodejs_plugin.py [hunk 2]
```diff
@@ -16,9 +16,9 @@
 import os
 
 import testscenarios
+import yaml as yaml_utils
```
Comment (kyrofa) [inline tests/integration/plugins_nodejs/test_nodejs_plugin.py]: Would it be a lot more work to change away from the name "yaml_utils" to "yaml" so future confusion is less likely?

Comment (sergiusens) [reply tests/integration/plugins_nodejs/test_nodejs_plugin.py]: this one is going away in a new PR. This is still useful when running in-source

File: tests/integration/lifecycle/test_snap.py [hunk 3]
```diff
@@ -91,24 +83,24 @@ def test_snap(self):
...
-                    command: snap/command-chain/snapcraft-runner $SNAP/command-assemble-service.wrapper
+                    command: command-assemble-service.wrapper
                     daemon: simple
-                    stop-command: snap/command-chain/snapcraft-runner $SNAP/stop-command-assemble-service.wrapper
+                    stop-command: stop-command-assemble-service.wrapper
                   binary-wrapper-none:
                     command: subdir/binary3
                   binary2:
-                    command: snap/command-chain/snapcraft-runner $SNAP/command-binary2.wrapper
+                    command: command-binary2.wrapper
```
Comment (kyrofa) [inline tests/integration/lifecycle/test_snap.py]: Given that legacy has its own version of this, would it be worth making it use a base? Or are you thinking that should involve a move to spread?

**Pull request merged**